### PR TITLE
Apply comments for PR 10985

### LINKED ISSUE #11072: Post-landing review comments for PR 10985
#### Problem
I didn't get to PR 10985 before it landed.  Looking at it now, I have the following questions/comments:

* ~~`StaticGroupsProvider::Decrypt` should take the `PacketHeader` as a const reference, not by value.~~ #11075 
* ~~`StaticGroupsProvider::Decrypt` should almost certainly take the `PacketBufferHandle` as an lvalue reference, not an rvalue reference, since it's not taking ownership of the buffer, I assume.~~ #11075 
* In `SecureUnicastMessageDispatch` all the MCSP stuff should go away; we should never be in a situation where `!state->GetSessionMessageCounter().GetPeerMessageCounter().IsSynchronized()` is true for a unicast message, and if that ever does happen we should just error out.  That whole block needs to move into `SecureGroupMessageDispatch` somewhere.
* In `SecureGroupMessageDispatch` if `packetHeader.IsSecureSessionControlMsg()` we should probably just error out.  The only control messages are MCSP messages, which are not group messages. [EDIT: This is not true.  `SecureGroupMessageDispatch` is called for a group session, and MCSP messages _do_ have a group session.  What they do not have is a group id to filter possible decryption keys on, but we are not doing anything with group ids here yet.]

 #### Proposed Solution
Fix those.

@jepenven-silabs

### PR DESCRIPTION
**Pull request by jepenven-silabs**
#### Problem
Some issues needed to be fixed following the merge of PR #10985 
See #11072 for full details.

#### Change overview
As discussed with @bzbarsky-apple apply the requested changes

#### Testing
Tested with Unit test.
Unit test for MCSP Message Header state added (session Group ID with a Destination Node ID)

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: cleanu PR approved by original requester.

**Review by jmartinez-silabs — APPROVED** [review]

**Review by rcasallas-silabs — APPROVED** [review]

**Comment by jepenven-silabs** [comment]
> @jepenven-silabs - it seems latest changes broke compilation.

Strange, since it was only a rebase on master with a small conflict fix. Working on it

File: src/transport/SessionManager.cpp [hunk 1]
```diff
@@ -502,6 +463,24 @@ void SessionManager::SecureGroupMessageDispatch(const PacketHeader & packetHeade
 
     VerifyOrExit(!msg.IsNull(), ChipLogError(Inet, "Secure transport received NULL packet, discarding"));
 
+    // MCSP check
+    if (packetHeader.IsSecureSessionControlMsg())
+    {
+        if (packetHeader.GetDestinationNodeId().HasValue() && packetHeader.HasPrivacyFlag())
+        {
+            // TODO
```
Comment (electrocucaracha) [inline src/transport/SessionManager.cpp:471]: It should be nice to include more information in this TODO comment

Comment (jepenven-silabs) [reply src/transport/SessionManager.cpp:471]: I thought that it was clear enough with the `//MCSP processing` a few line below.
I'll add the comment next time I fiddle with this section of code.

**Pull request merged**